src/pages/taskQueue/TaskQueueTable.tsx

```diff
@@ -72,18 +72,30 @@ export const TaskQueueTable = () => {
       key: "displayName",
       className: "cy-task-queue-col-task",
       width: "30%",
-      render: (_, { displayName, id, project, buildVariant }) => (
+      render: (_, { displayName, id, project, buildVariant, requester }) => (
         <TaskCell>
           <Body>
-            <StyledRouterLink
-              data-cy="current-task-link"
-              to={getTaskRoute(id)}
-              onClick={() =>
-                taskQueueAnalytics.sendEvent({ name: "Click Task Link" })
-              }
-            >
-              {displayName}
-            </StyledRouterLink>
+            {requester === TaskQueueItemType.Patch ? (
+              <StyledRouterLink
+                data-cy="current-task-link"
+                to={getTaskRoute(id)}
+                onClick={() =>
+                  taskQueueAnalytics.sendEvent({ name: "Click Task Link" })
+                }
+              >
+                {displayName}
+              </StyledRouterLink>
+            ) : (
+              <StyledLink
+                data-cy="current-task-link"
+                href={`${getUiUrl()}/task/${id}`}
```

Comment:
we should make this a shared url generator since we use the same URL structure here: https://github.com/evergreen-ci/spruce/blob/master/src/pages/host/Metadata.tsx#L29